programmers/모의고사.js

```diff
@@ -0,0 +1,46 @@
+function solution(answers) {
+    const answerArr = []
+    const supo1 = [1,2,3,4,5];
+    const supo2 = [2,1,2,3,2,4,2,5];
+    const supo3 = [3,3,1,1,2,2,4,4,5,5];
+    const rank = [0,0,0]; //result[0] -> supo1
+    let i = 0; 
+    let j = 0; 
+    let k = 0;
+    
+    for (let answer of answers) {
+        if (answer === supo1[i]) {
+            rank[0]++;
+            i++;
+            if (i === 5) i = 0;
+        }
+        else {
+            i++;
+            if (i === 5) i = 0; 
+        }
+        
+        if (answer === supo2[j]) {
+            rank[1]++;
+            j++;
+            if (j === 8) j = 0;
+        }
+        else {
+            j++;
+            if (j === 8) j = 0;
+        }
+        
+        if (answer === supo3[k]) {
+            rank[2]++;
+            k++;
+            if (k === 10) k = 0;
+        }
+        else {
+            k++;
+            if (k === 10) k = 0;
+        }        
+    }    
+    for (let l = 0; l < 3; l++) {
+        if (rank[l] === Math.max.apply(null, rank)) answerArr.push(l+1)
```

Comment:
```apply```로 할 수는 있으나 아래처럼 전개 연산자를 사용하는 것이 코드가 더 간결해집니다.

변수명뒤에 Arr, Str, Data, Val 등과 같은 이름은 붙이지 않는 것이 좋습니다.
배열은 복수형으로 선언해주세요.

배열 선언은 ```const```로 잘 해주셨으나 추가적인 지식에 대해 알려드릴게요.

배열 같은 경우는 iterable 객체라고 합니다.
iterable 객체는 배열, 오브젝트 등의 타입들을 통틀어 이야기 하는데요.
즉, 내부 원소들이 존재하는 객체들을 iterable 객체 라고 합니다.
Array, Object, Set, Map 등이 이에 해당하겠네요.

iterable 객체들의 특징은 mutable 하다는 것인데요.
mutable의 사전적 정의는 '변이 가능한' 입니다.
즉, 내부 원소들이 변할 수 있다는 것이지요.
따라서 iterable 객체들은 무조건 const로 선언하는 것이 관례입니다.
특수한 경우에만 let을 선언해주고 있어요.

예를 들어, 내부 원소를 바꾸는 것이 아니라 배열을 통째로 변경해야하는 경우에는 let을 선언해줍니다. 아래처럼요.

let arr = [1, 2, 3];
arr = ["1", "2", "3"];
하지만, 웬만해서는 위처럼 배열 전체를 바꾸는 행위는 지양해야 합니다.
협업할 때 누군가의 실수로 값을 바꿀 수도 있기 때문이죠.
이와 같은 이유로 let보다는 const를 사용하는 것입니다.
```suggestion
        if (rank[l] === Math.max.(...rank)) answers.push(l+1)
```